resteasy-client-vertx/src/main/java/org/jboss/resteasy/client/jaxrs/engines/vertx/VertxClientHttpEngine.java

```diff
@@ -0,0 +1,309 @@
+            CompletableFuture<T> tmp = new CompletableFuture<>();
+            vertx.executeBlocking(promise -> {
+                try {
+                    T result = extractor.extractResult(response);
+                    tmp.complete(result);
+                } catch (Exception e) {
+                    tmp.completeExceptionally(e);
+                }
+            }, ar -> {
+                //
+            });
+            return tmp;
+        });
+        if (callback != null) {
+            future = future.whenComplete((response, throwable) -> {
+                if (throwable != null) {
+                    callback.failed(throwable);
+                } else {
+                    callback.completed(response);
+                }
+            });
+        }
+        return future;
+    }
+
+    @Override
+    public <T> CompletableFuture<T> submit(final ClientInvocation request,
+                                           final boolean buffered,
+                                           final ResultExtractor<T> extractor,
+                                           final ExecutorService executorService) {
+        return submit(request).thenCompose(response -> {
+            CompletableFuture<T> tmp = new CompletableFuture<>();
+            executorService.execute(() -> {
+                try {
+                    T result = extractor.extractResult(response);
+                    tmp.complete(result);
+                } catch (Exception e) {
+                    tmp.completeExceptionally(e);
+                }
+            });
+            return tmp;
+        });
+    }
+
+    public CompletableFuture<ClientResponse> submit(final ClientInvocation request) {
+
+        HttpMethod method;
+        try {
+            method = HttpMethod.valueOf(request.getMethod());
+        } catch (IllegalArgumentException e) {
+            method = HttpMethod.OTHER;
+        }
+
+        Object entity = request.getEntity();
+        Buffer body;
+        if (entity != null) {
+            body = Buffer.buffer(requestContent(request));
+        } else {
+            body = null;
+        }
+
+        RequestOptions options = new RequestOptions();
+        MultiMap headers = MultiMap.caseInsensitiveMultiMap();
+        request.getHeaders().asMap().forEach(headers::add);
+        options.setHeaders(headers);
+        if (body != null) {
+            headers.set(HttpHeaders.CONTENT_LENGTH, "" + body.length());
+        }
+        options.addHeader(HttpHeaders.USER_AGENT.toString(), "Vertx");
+
+        URI uri = request.getUri();
+        options.setHost(uri.getHost());
+        options.setPort(uri.getPort());
+        options.setURI(uri.getRawPath());
+
+        CompletableFuture<ClientResponse> future = new CompletableFuture<>();
+
+        HttpClientRequest clientRequest = httpClient.request(method, options, response -> {
+            future.complete(toRestEasyResponse(request.getClientConfiguration(), response));
```

Comment:
the deprecation advocates to use Vert.x Web Client (which uses this HttpClient under the hood) but I think for this low level client using the Vert.x HttpClient is more appropriate (less dependencies / less overhead). This deprecation is temporary until Vert.x 4 that gets the `HttpClient` methods un-deprecated (but signature changes a bit).

I can add a suppress warnings annotation if you want with a comment saying that this will go away with Vert.x 4

Reply:
Thanks